.github/workflows/release-publish.yaml

```diff
@@ -20,7 +20,7 @@ jobs:
           # Note: we fetch the tags here instead of using actions/checkout's "fetch-tags"
           # because of https://github.com/actions/checkout/issues/1467
           git fetch --force --tags --depth 1
-          git describe --dirty --long --match '[0-9]*.[0-9]*.[0-9]*' --exclude '*[^0-9.]*'
+          git describe --dirty --tags --long --match '[0-9]*.[0-9]*.[0-9]*' --exclude '*[^0-9.]*'
```

Comment:
I think the problem here is not that the command is wrong, but that the `0.2.0` tag is not annotated. Quoting `git tag --help`:

> Annotated tags are meant for release while lightweight tags are meant for private or temporary object labels. For this reason, some git commands for naming objects (like git describe) will ignore lightweight tags by default.

If you create an annotated tag it should work:

```
❯ git describe --dirty --long --match '[0-9]*.[0-9]*.[0-9]*' --exclude '*[^0-9.]*'
fatal: No annotated tags can describe 'c7149c3b1d76e4284be740f9ce5031e8f3cdf9ef'.
However, there were unannotated tags: try --tags.
❯ git tag --annotate --message="test" 9.9.9
❯ git describe --dirty --long --match '[0-9]*.[0-9]*.[0-9]*' --exclude '*[^0-9.]*'
9.9.9-0-gc7149c3
```

Ideally, the tag should be also signed (`--sign`) to offer some more confidence on the provenance, but this part shouldn't matter for this release workflow.

Reply:
So you can't do this from within  a GitHub release?

Reply:
Do what exactly?

Reply:
Oh I think I see what you mean; no, you can't create the tag through a Github release because Github [only creates lightweight tags](https://github.com/orgs/community/discussions/4924)  unfortunately. The good news is that this release workflow will also create the Github release.

Reply:
Understood. Thanks!